src/resource-manager-session.c

```diff
@@ -0,0 +1,137 @@
+/* SPDX-License-Identifier: BSD-2 */
+#include <glib.h>
+#include <inttypes.h>
+
+#include "access-broker.h"
+#include "resource-manager.h"
+#include "resource-manager-session.h"
+#include "session-list.h"
+#include "session-entry.h"
+#include "tabrmd.h"
+#include "tpm2-command.h"
+#include "tpm2-header.h"
+#include "tpm2-response.h"
+#include "util.h"
+
+/*
+ * Load the TPMS_CONTEXT associated with the provided SessionEntry*.
+ * This function will always return a Tpm2Response object (never NULL)
+ * though the response object may indicate a failure in the response code.
+ */
+Tpm2Response*
+load_session (ResourceManager *resmgr,
+              SessionEntry *entry)
+{
+    Tpm2Command *cmd = NULL;
+    Tpm2Response *resp = NULL;
+    size_buf_t *size_buf = NULL;
+    TSS2_RC rc = TSS2_RC_SUCCESS;
+
+    g_debug ("%s: SessionEntry 0x%" PRIxPTR, __func__, (uintptr_t)entry);
+    size_buf = session_entry_get_context (entry);
+    cmd = tpm2_command_new_context_load (size_buf->buf, size_buf->size);
+    if (cmd == NULL) {
+        g_critical ("%s: failed to allcoate ContextLoad Tpm2Command",
+                    __func__);
+        resp = tpm2_response_new_rc (NULL, TSS2_RESMGR_RC_GENERAL_FAILURE);
+        goto out;
+    }
+    resp = access_broker_send_command (resmgr->access_broker, cmd, &rc);
+    if (rc != TSS2_RC_SUCCESS) {
+        g_critical ("%s: TCTI failed while loading session context from "
+                   "SessionEntry 0x%" PRIxPTR ", got RC 0x%" PRIx32,
+                   __func__, (uintptr_t)entry, rc);
+        resp = tpm2_response_new_rc (NULL, rc);
```

Comment:
We reassign resp here, are we leaking the one returned from the `access_broker_send_command()` call above?

Reply:
You'll need to take a look at the `access_broker_send_command` function to understand this. The `Tpm2Response` object returned from the function represents the response from the TPM. The `rc` parameter (an 'out' value / passed by reference) represents the `TPM2_RC` from the TCTI layer. If the TCTI returns an error then the return value will be NULL. Thus the 'resp' value here will not be overwritten / leaked because it is NULL.